### PR TITLE
Add ability to set video encoder bitrate based on source video stream bit rate

### PR DESCRIPTION
**Pull request by johnyerhot**
Did some work to support a `bt-ratio` configuration option. This value is basically multiplied by the source file's video bit rate to set a new encoding bit rate for the video source.

My use case is mainly around converting my massive video library to hevc where I want to be able to encode at 60% of the source streams bit rate to get basically the same quality since hevc compression is so much better. I'd rather not use `-qp` or `-crf`and have had better results using this, esp vs. setting different `crf-profiles`.

Few notes:

* Setting a `crf` value will still always use `crf` or `qp` for hevc.
* setting a `crf-profile` will override this.
* `max-bitrate` should still kick in (I think)
* basically all `crf` options override `bt-ratio`.

I'm not a Python expert so unsure if my code could be better but Works on my Machine™ 😄

### CODE REVIEW

**Comment by mdhiggins** [comment]
Been very busy with work so haven't had a chance to address this directly.

Interesting concept but I think before merge I would like to make a few changes. What I think would be beneficial for this feature would be a dict type setting where you can specify source codecs and multipliers for each so that it's a little more dynamic depending on the source material

Setting sample would look something like:
`bitrate-ratio = h264:0.6, divx:0.3`

I would also shift the bitrate adjustment out of the converter and just calculate the adjusted bitrate rather than passing a multiplier and bitrate, but that's more stylistic than anything but it leaves the converter code alone. When I get a chance I'll put together a test and link it here

**Comment by mdhiggins** [comment]
Alright finally got around to this

https://github.com/mdhiggins/sickbeard_mp4_automator/tree/bitrate-ratio

There's a branch for testing that has the new `bitrate-ratio` setting under video

You can specify a dict with key:float values. The default ratio will be 1.0

Here is a sample setting

`bitrate-ratio = h264:0.6, hevc:0.8`

This allows you to have a different ratio depending on the incoming source codec. Looks like its all working on my end but if you could test it out before I merge things into the master branch it would be appreciated. Bitrate settings still respect max-bitrate and all the CRF settings take priority if specified

**Comment by johnyerhot** [comment]
So awesome, thank you @mdhiggins 🎉

**Comment by mdhiggins** [comment]
Shall I take it to mean that's working for you? 👍 if so I'll merge it in and update the wiki

**Comment by johnyerhot** [comment]
Got a batch of videos I'll test it on shortly. Just waiting for a new ffmpeg to compile (don't ask :P).

**Comment by johnyerhot** [comment]
I initially got a little tripped up because I didn't notice that the `bitrate-ratio` diict was actually specifying the source codec and the ratio to use for that source. I had it the otherway around (always use this ratio when encoding with this encoder).

Your process makes a lot more sense than my original thought pattern as it allows for way more flexibility.

Anyways, yep. Worked great on this batch of videos. Thank you, @mdhiggins.

**Comment by mdhiggins** [comment]
Thanks for merging. Updated the wiki to reflect the new option. Gonna close this out since the feature is implemented
